### PR TITLE
Support thematic breaks

### PR DESCRIPTION
**Pull request by mportiz08**
Bug/issue #, if applicable: 125507297

## Summary

Adds support for rendering "thematic breaks" within content using the `<hr>` element.

With the corresponding Swift-DocC PR, the markdown syntax for horizontal rules can be used in content to indicate a thematic break within content, and this PR adds rendering support for this.

**Example Markdown:**

```diff
diff --git a/SwiftDocCRender.docc/contributing/Internals.md b/SwiftDocCRender.docc/contributing/Internals.md
index d3054747..2500a12d 100644
--- a/SwiftDocCRender.docc/contributing/Internals.md
+++ b/SwiftDocCRender.docc/contributing/Internals.md
@@ -6,6 +6,8 @@ Learn how DocC-Render is built internally, the principles behind its structure,
 
 DocC-Render supports rendering reference pages, tutorials, and articles to document your project in a structured manner.
 
+---
+
 Each page fetches its required render JSON asynchronously, from the `/data` directory. To learn more about the types of pages DocC-Render supports, visit the [DocC Docs](https://www.swift.org/documentation/docc).
 
 ### Modular Architecture
```

**Example rendering:**

Screenshot 2024-03-25 at 12 19 20 PM

## Dependencies

https://github.com/apple/swift-docc/pull/865

## Testing

Steps:
1. Checkout/build the branch of Swift-DocC from [Support thematic breaks #865](https://github.com/apple/swift-docc/pull/865)
2. Checkout/build the branch of Swift-DocC-Render from this PR
3. Add/modify content with a markdown horizontal rule (like the example above)
4. Preview the docs using `DOCC_HTML_DIR=/path/to/swift-docc-render/dist swift run docc preview ExampleContent.docc` and verify that a horizontal rule is rendered in the appropriate place

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran `npm test`, and it succeeded
- [X] Updated documentation if necessary

### CODE REVIEW

**Comment by mportiz08** [comment]
@swift-ci test

**Review by emilyychenn — COMMENTED** [review]
This works great when I test it locally with my version of swift-docc (from [here](https://github.com/apple/swift-docc/pull/865))! Just curious, is the thematic break supposed to have more emphasis than the horizontal line between the "Overview" and "Topics" sections?

Screenshot 2024-03-26 at 13 15 01

**Review by emilyychenn — APPROVED** [review]

**Review by marinaaisa — CHANGES_REQUESTED** [review]
I tested it and it works great!
I added a comment to fix how `hr` look by default.

Btw, you missed the `docc` executable in the testing instructions, if you want to update them:
```
DOCC_HTML_DIR=/path/to/swift-docc-render/dist swift run docc preview ExampleContent.docc
```

**Comment by mportiz08** [comment]
> This works great when I test it locally with my version of swift-docc (from [here](https://github.com/apple/swift-docc/pull/865))! Just curious, is the thematic break supposed to have more emphasis than the horizontal line between the "Overview" and "Topics" sections?
> 
> Screenshot 2024-03-26 at 13 15 01

No, it should probably look the same. Thanks for catching that!

File: src/components/ContentNode/ThematicBreak.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<template>
+  <hr class="thematic-break" />
+</template>
+
+<style scoped lang="scss">
+@import 'docc-render/styles/_core.scss';
+
+.thematic-break {
+  @include space-out-between-siblings(var(--spacing-stacked-margin-xlarge));
+  border-color: var(--color-grid, currentColor);
```
[the remark this answers is not in the document]
Comment (mportiz08) [reply src/components/ContentNode/ThematicBreak.vue]: Fixed with 261e2d06c3ee8ff5b1af92856675f3060a7204bd

**Comment by mportiz08** [comment]
@swift-ci test

**Comment by mportiz08** [comment]
> > This works great when I test it locally with my version of swift-docc (from [here](https://github.com/apple/swift-docc/pull/865))! Just curious, is the thematic break supposed to have more emphasis than the horizontal line between the "Overview" and "Topics" sections?
> > Screenshot 2024-03-26 at 13 15 01
> 
> No, it should probably look the same. Thanks for catching that!

Fixed

Screen Shot 2024-03-26 at 09 30 33

**Review by marinaaisa — APPROVED** [review]
LGTM!

**Pull request merged**